File: LeetCode/Medium/0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        int left = 0;
        int right = height.length - 1;
        int ans = 0;

        while(left != right) {
            // System.out.println("left : " + left + " / right : " + right);
            ans = Math.max(
                ans,
                this.calc(height[left], height[right], right - left)
            );

            if(height[left] < height[right]) left++;
            else right--;
        }
        // System.out.println("끝");

        return ans;
    }

    private int calc(int leftV, int rightV, int length) {
        return Math.min(leftV, rightV) * length;
    }
}

File: LeetCode/Medium/0011-container-with-most-water/0011-container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        
        var left = 0
        var right = height.size - 1
        var ans = 0

        while(left != right) {
            ans = Math.max(
                ans,
                this.calc(height[left], height[right], right - left)
            )

            if(height[left] < height[right]) left++
            else right--
        }

        return ans
    }

    fun calc(leftV: Int, rightV: Int, length: Int): Int {
        return Math.min(leftV, rightV) * length
    }
}